sklearn/utils/validation.py

```diff
@@ -1361,6 +1361,59 @@ def _allclose_dense_sparse(x, y, rtol=1e-7, atol=1e-9):
                      "matrix and an array")
 
 
+def _check_precomputed_gram_matrix(X, precompute, X_offset, X_scale,
+                                   rtol=1e-7,
+                                   atol=1e-5):
```

Comment:
```suggestion
def _check_precomputed_gram_matrix(
    X, precompute, X_offset, X_scale, rtol=1e-7, atol=1e-5
):
```

Reply:
If we are not using the utility elsewhere, I would move it to the top of the file where it is called.
We would need to have a small unit test specifically for this function.